app/src/features/projects/components/ProjectFundingForm.test.tsx

```diff
@@ -95,4 +95,152 @@ describe('ProjectFundingForm', () => {
 
     expect(asFragment()).toMatchSnapshot();
   });
+
+  it('shows add funding source dialog on add click', async () => {
+    const existingFormValues: IProjectFundingForm = {
+      funding_sources: [
+        {
+          id: 11,
+          agency_id: 1,
+          investment_action_category: 1,
+          investment_action_category_name: 'action 1',
+          agency_project_id: '111',
+          funding_amount: 222,
+          start_date: '2021-03-14',
+          end_date: '2021-04-14',
+          revision_count: 23
+        },
+        {
+          id: 12,
+          agency_id: 2,
+          investment_action_category: 2,
+          investment_action_category_name: 'category 1',
+          agency_project_id: '112',
+          funding_amount: 223,
+          start_date: '2021-03-15',
+          end_date: '2021-04-15',
+          revision_count: 24
+        },
+        {
+          id: 13,
+          agency_id: 3,
+          investment_action_category: 3,
+          investment_action_category_name: 'not applicable',
+          agency_project_id: '113',
+          funding_amount: 224,
+          start_date: '2021-03-16',
+          end_date: '2021-04-16',
+          revision_count: 0
+        }
+      ]
+    };
+
+    const { asFragment, getByTestId, queryByText } = render(
+      <Formik
+        initialValues={existingFormValues}
+        validationSchema={ProjectFundingFormYupSchema}
+        validateOnBlur={true}
+        validateOnChange={false}
+        onSubmit={async () => {}}>
+        {() => <ProjectStepComponents component="ProjectFunding" codes={codes} />}
+      </Formik>
+    );
+
+    const addButton = await getByTestId('add-button');
+    expect(addButton).toBeInTheDocument;
+    fireEvent.click(addButton);
+
+    expect(await queryByText('Agency Details')).toBeInTheDocument;
+
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
I dont think this snapshot is actually working the way you expect
I would just remove it as I think the other assertions you have in place are sufficient for good quality tests :) or we need to fix this

Reply:
same as above